jdbc/pom.xml

```diff
@@ -32,6 +32,201 @@
+          <artifactId>hive-shims-0.23</artifactId>
+          <version>${hive2.version}</version>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <profile>
+      <id>jdbc-phoenix</id>
+      <properties>
+        <phoenix.version>4.7.0</phoenix.version>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.phoenix</groupId>
+          <artifactId>phoenix-core</artifactId>
+          <version>${phoenix.version}</version>
+        </dependency>
+
+        <dependency>
+          <groupId>xerces</groupId>
+          <artifactId>xercesImpl</artifactId>
+          <version>2.11.0</version>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <profile>
+      <id>jdbc-hadoop2</id>
+      <properties>
+        <hadoop-common.version>2.7.3</hadoop-common.version>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-common</artifactId>
+          <version>${hadoop-common.version}</version>
+          <exclusions>
+            <exclusion>
+              <groupId>com.sun.jersey</groupId>
+              <artifactId>jersey-core</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>com.sun.jersey</groupId>
+              <artifactId>jersey-json</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>com.sun.jersey</groupId>
+              <artifactId>jersey-server</artifactId>
+            </exclusion>
+
+            <exclusion>
+              <groupId>javax.servlet</groupId>
+              <artifactId>servlet-api</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>org.apache.avro</groupId>
+              <artifactId>avro</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>org.apache.jackrabbit</groupId>
+              <artifactId>jackrabbit-webdav</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>io.netty</groupId>
+              <artifactId>netty</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>commons-httpclient</groupId>
+              <artifactId>commons-httpclient</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>org.apache.zookeeper</groupId>
+              <artifactId>zookeeper</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>org.eclipse.jgit</groupId>
+              <artifactId>org.eclipse.jgit</artifactId>
+            </exclusion>
+            <exclusion>
+              <groupId>com.jcraft</groupId>
+              <artifactId>jsch</artifactId>
+            </exclusion>
+          </exclusions>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <profile>
+      <id>jdbc-hadoop3</id>
+      <properties>
+        <hadoop-common.version>3.0.0</hadoop-common.version>
```

Comment:
shouldn't hadoop2 vs hadoop3 be using the hadoop profile

Reply:
I initially thought about it, but then changed, as there can be a case where only Postgres or Hive without Hadoop is required and because hadoop2/3 is specified for zeppelin-zengine it will download Hadoop dependencies for JDBC as well resulting in increased size of JDBC. or do you mean instead of using jdbc-hadoop2/jdbc-hadoop3 there should only be jdbc-hadoop?